### PR TITLE
add: check if output is local when generating stage name

### PR DESCRIPTION
**Pull request by efiop**
When adding a remote file with

```
$ dvc add s3://bucket/path
```

it results in

```
Error: failed to add file - is local is not supported for s3
```

Signed-off-by: Ruslan Kuprieiev <ruslan@iterative.ai>

### CODE REVIEW

File: dvc/stage.py [hunk 1]
```diff
@@ -339,6 +339,7 @@ def _stage_fname(cls, fname, outs, add):
     def _expand_to_path_on_add_local(add, fname, out, path_handler):
         if (
             add
+            and out.scheme == "local"
             and out.is_local
```
Comment (mroutis) [inline dvc/stage.py]: @efiop , the `out.is_local` shouldn't check this? if it is deprecated, let's remove this line

Comment (efiop) [reply dvc/stage.py]: @mroutis `is_local` is about local output being inside of a dvc repo. But yeah, makes sense to make all non-local outputs be outside of repo. I'll rename it to be less confusing as well. Thanks for pointing that out! :slightly_smiling_face:

**Comment by efiop** [comment]
@mroutis changed it in this PR. Please take a look. :slightly_smiling_face:

**Comment by efiop** [comment]
@mroutis Thanks for the feedback! :slightly_smiling_face:

**Pull request merged**